.github/workflows/ci-cd.yml

```diff
@@ -22,54 +22,91 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  CI-CD:
+  lint-check:
     runs-on: ubuntu-latest
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - name: Setup Nodejs 16.x
-        uses: actions/setup-node@v3
-        with:
-          node-version: 16
-      - name: install
-        run: yarn --frozen-lockfile
-      # waiting for approval - uses: nrwl/nx-set-shas@v3
-      - name: set base branch
-        run: |
-          if [ $BRANCH_REF = 'refs/heads/main' ]; then 
-            echo 'BASE_BRANCH=origin/main~1' >> $GITHUB_ENV
-          else
-            echo 'BASE_BRANCH=origin/main' >> $GITHUB_ENV
-          fi
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
       - name: lint & check
         run: |
           yarn nx workspace-lint
           yarn nx format:check
           git fetch origin main
           yarn nx affected --target=lint --base=${{ env.BASE_BRANCH }} --parallel=3
-      - name: build & test
+  test-publisher:
+    needs: lint-check
+    uses: ./.github/workflows/test-suite.yml
+    with:
+      suite_name: test-publisher
+      branch_ref: $BRANCH_REF
+    secrets:
```

Comment:
Seems like all of the publish_token and account_id are same in test-suit workflow, do we still need to pass them as secrets parameters?  Could we move them into test-suit workflow?

Reply:
@vincentsong We cannot use the secrets directly wit the resuable jobs.